### PR TITLE
feat/CE-187-New dropdown available without clearing cache

### PR DESCRIPTION
**Pull request by Scarlett-Truong**
# Description

Please provide a summary of the change and the issue fixed. Please include relevant context. List dependency changes.

Fixes # (issue)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Further comments

---

Thanks for the PR!

Any successful deployments (not always required) will be available below.
[Backend](https://nr-compliance-enforcement-315-backend.apps.silver.devops.gov.bc.ca/) available
[Frontend](https://nr-compliance-enforcement-315-frontend.apps.silver.devops.gov.bc.ca/) available

Once merged, code will be promoted and handed off to following workflow run.
[Main Merge Workflow](https://github.com/bcgov/nr-compliance-enforcement/actions/workflows/merge-main.yml)

### CODE REVIEW

**Comment by afwilcox** [comment]
@Scarlett-Truong  - This looks good so far, the only thing I see is that I think you need to use a Redux migration for the addition of the code table version into the state to ensure that it gets loaded properly the first time.

I'm going to do a bit more testing, but so far it's working exactly as I would have expected it to.

@marqueone-ps  can answer any questions about it if you need it.

File: backend/src/v1/configuration/configuration.controller.ts [hunk 1]
```diff
@@ -22,8 +24,32 @@ export class ConfigurationController {
...
+    @Token() token) 
+  {
+    const result = await this.configurationService.findOne(configurationCode);
+
+    //If configuration is code table version, call another case managment api
+    if(configurationCode === 'CDTABLEVER') {
+      const { data } = await get(token, { 
+        query : '{getConfigurationCode(configuration_code: "CDTABLEVER"){configuration_code configuration_value  active_ind}}'
+      });
+      const caseData = {
```
Comment (marqueone-ps) [inline backend/src/v1/configuration/configuration.controller.ts:38]: @Scarlett-Truong you might want to check that result is both not null and that there are elements in the results array. There may be a scenario where that can cause an error when creating complaintData.

File: frontend/src/app/store/reducers/app.ts [hunk 2]
```diff
@@ -389,6 +414,10 @@ const initialState: AppState = {
   configurations: {
     configurations: undefined,
   },
+  codeTableVersion: (localStorage.getItem("codeTableVersion")!== null) ?
```
Comment (marqueone-ps) [inline frontend/src/app/store/reducers/app.ts:417]: @Scarlett-Truong is this necessary? Is there a reason the codeTableVersion would live outside the scope of redux and have a separate localState property for tracking the codeTableVersion?

File: backend/src/v1/configuration/configuration.controller.ts (hunk 1 above)
Comment (Scarlett-Truong) [reply backend/src/v1/configuration/configuration.controller.ts:38]: yeah, good point 😃

File: frontend/src/app/store/reducers/app.ts (hunk 2 above)
Comment (marqueone-ps) [reply frontend/src/app/store/reducers/app.ts:417]: Thats why I'm a bit confused by this. You're checking the localState for that value and setting it to the initialState, but if that value isn't there you're defaulting it to ```{ complaintManagement: "1", caseManagement: "1" } ``` and then once you're request for the codeTables is complete you're dispatching the version numbers to redux. The localStorage step seems redundant.

**Pull request merged**